finesse/gui/temp_control.py

```diff
@@ -0,0 +1,312 @@
+"""Panel and widgets related to temperature monitoring."""
+from datetime import datetime
+from functools import partial
+from importlib import resources
+from typing import Tuple
+
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+from PySide6.QtCore import Qt
+from PySide6.QtGui import QImage, QPixmap
+from PySide6.QtWidgets import (
+    QGridLayout,
+    QGroupBox,
+    QLabel,
+    QLineEdit,
+    QProgressBar,
+    QPushButton,
+    QSpinBox,
+    QWidget,
+)
+
+poll_on_img_data = resources.read_binary("finesse.gui.images", "poll_on.png")
+poll_off_img_data = resources.read_binary("finesse.gui.images", "poll_off.png")
+alarm_on_img_data = resources.read_binary("finesse.gui.images", "alarm_on.png")
+alarm_off_img_data = resources.read_binary("finesse.gui.images", "alarm_off.png")
```

Comment:
It seems that `read_binary` [is deprecated since Python 3.11](https://docs.python.org/3/library/importlib.resources.html#importlib.resources.read_binary).  

I'm sure you have considered this - and possibly discarded it - but I honestly think it is cleaner to just use the relative location of the images folder with respect to this one and open an IO stream:

```suggestion
images = Path(__file__).parent / "images"
poll_on_img_data = open(images / "poll_on.png", "rb")
poll_off_img_data = open(images /  "poll_off.png", "rb")
alarm_on_img_data = open(images /  "alarm_on.png", "rb")
alarm_off_img_data = open(images /  "alarm_off.png", "rb")
```
(Or something along this lines. I'm not totally sure if this open the file in the correct mode)

This will also save the issue with the hidden imports in Pyinstaller.